### PR TITLE
[4.2] - RavenDB-15888 - NRE upon database creation

### PR DESCRIPTION
**Pull request by grisha-kotler**

### CODE REVIEW

File: src/Raven.Server/ServerWide/ClusterStateMachine.cs [hunk 1]
```diff
@@ -476,12 +476,23 @@ protected override void Apply(TransactionOperationContext context, BlittableJson
                         throw new UnknownClusterCommand(massage);
                 }
 
+                if (_parent == null)
+                    throw new InvalidOperationException("TEST");
```
Comment (ppekrol) [inline src/Raven.Server/ServerWide/ClusterStateMachine.cs]: test?

Comment (grisha-kotler) [reply src/Raven.Server/ServerWide/ClusterStateMachine.cs]: done

File: src/Raven.Server/ServerWide/ClusterStateMachine.cs [hunk 2]
```diff
@@ -510,6 +524,40 @@ protected override void Apply(TransactionOperationContext context, BlittableJson
                     LeaderShipDuration = leader?.LeaderShipDuration,
                 });
             }
+
+            void DismissUnrecoverableNotification()
+            {
+                try
+                {
+                    serverStore.NotificationCenter.Dismiss(AlertRaised.GetKey(AlertType.UnrecoverableClusterError, null), context.Transaction, false);
```
Comment (karmeli87) [inline src/Raven.Server/ServerWide/ClusterStateMachine.cs]: Let's make the key here more unique (using `term/raft-index`), so I'll will add/dismiss the proper `UnrecoverableClusterError` alert

Comment (grisha-kotler) [reply src/Raven.Server/ServerWide/ClusterStateMachine.cs]: done

File: src/Raven.Server/ServerWide/ClusterStateMachine.cs [hunk 3]
```diff
@@ -510,6 +524,41 @@ protected override void Apply(TransactionOperationContext context, BlittableJson
                     LeaderShipDuration = leader?.LeaderShipDuration,
                 });
             }
+
+            void DismissUnrecoverableNotification()
+            {
+                try
+                {
+                    serverStore.NotificationCenter.Dismiss(AlertRaised.GetKey(AlertType.UnrecoverableClusterError, $"{_parent.CurrentTerm}/{index}"), context.Transaction, false);
```
Comment (ppekrol) [inline src/Raven.Server/ServerWide/ClusterStateMachine.cs]: `false` - use named parameter

Comment (grisha-kotler) [reply src/Raven.Server/ServerWide/ClusterStateMachine.cs]: done

**Pull request merged**